duration.go

```diff
@@ -0,0 +1,161 @@
+package null
+
+import (
+	"database/sql/driver"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"reflect"
+	"strconv"
+	"strings"
+	"time"
+)
+
+var (
+	errBadDurationSQLType = errors.New("incompatible SQL type for Scan() into null.Duration")
+)
+
+// Duration is a nullable time.Duration. It supports SQL and JSON serialization.
+// It will marshal to null if null.
+type Duration struct {
+	Duration time.Duration
+	Valid    bool
+}
+
+// Scan implements the Scanner interface.
+// Supports the ´interval` type for PostgreSQL
+func (d *Duration) Scan(value interface{}) error {
```

Comment:
It would be useful if you could document the format this supports. Right now it's very specific in accepting only `hour:minute:seconds`, which may be surprising as `minute:seconds` and `hour:minute` are also common.